src/root.zig

```diff
@@ -165,15 +167,19 @@ pub fn AlgorithmType(
             };
         }
 
-        fn resize(self: *Self, new_cols: usize, new_rows: usize) !void {
+        /// Resize matrixs and buffer if it is needed
+        pub fn resizeIfNeeded(self: *Self, max_haystack: usize, max_needle: usize) !void {
```

Comment:
The idea was that `resizeIfNeeded` can do fine optimisations, and `resize` is a hard command.
`resizeIfNeeded` can't just call `resize`, because `resize` resize all buffers and matrix.
So what you want is that just keep the `resizeIfNeeded` with the name `resize` ?
The advantage of the hard `resize` was that you can shrink the memory used. But it is maybe a bad use case.

Reply:
Okay I understand what you mean now. Yes that's fine. I might combine them into an `ensureSize` function instead of a `resize` then that just bumps the upper limit but never shrinks.